crates/shared/src/event_handling.rs

```diff
@@ -30,7 +30,7 @@ where
     block_retriever: B,
     contract: C,
     store: S,
-    last_handled_block: Option<BlockNumberHash>,
+    last_handled_blocks: Vec<BlockNumberHash>,
```

Comment:
Because we append to the end and pop from both sides, we should use a `VecDeque` (you can use `pop_front` to remove the front element).

Just a minor suggestion, using the a Vec is also fine.

Reply:
I would have to call `pop_front` multiple times for reorg depth > 1:
```
while last_handled_blocks.len() > MAX_REORG_BLOCK_COUNT {
   self.last_handled_blocks = self.last_handled_blocks.pop_front();
}
```

making the code less clear, in my humble opinion. Regarding performance, I guess no major difference on a 25/64 element list. Actually, VecDeque in this case has more copies to do than Vec (?)

Reply:
Ah, right. I thought there might have been a "truncate_front" or similar API, but that doesn't seem to exist.